lib/livebook_web/router.ex

```diff
@@ -16,6 +16,17 @@ defmodule LivebookWeb.Router do
     plug LivebookWeb.UserPlug
   end
 
+  pipeline :js_output_assets do
+    plug :put_secure_browser_headers
+  end
+
+  scope "/", LivebookWeb do
+    pipe_through [:js_output_assets]
+
+    get "/sessions/assets/:hash/*file_parts", SessionController, :show_cached_asset
+    get "/sessions/:id/assets/:hash/*file_parts", SessionController, :show_asset
```

Comment:
I am thinking we should keep it and only allow `:file_part` with no nesting. I know you are checking to see if we are not doing path traversal (and we should keep it) but if we can avoid the attack vector altogether, then that's definitely better.

Reply:
Ignore me. We already prefix the path by hash, we should be fine.

Reply:
The assets may have subdirectories, so we want to support nested paths :)